comm.cpp

```diff
@@ -245,6 +245,17 @@ void Endpoint::print_statistics()
            _write_total);
 }
 
+int Endpoint::write_msg(const mavlink_message_t *msg)
+{
+    struct buffer buffer;
+    uint8_t data[MAVLINK_MAX_PACKET_LEN];
+
+    buffer.data = data;
+    buffer.len = mavlink_msg_to_send_buffer(data, msg);
```

Comment:
we should avoid this data copy... you will actually copy 3 times if you do this (from buffer to msg, from msg to temporary buffer, from temporary buffer to send buffer).... a router should be smarter than that.